Makefile

```diff
@@ -386,3 +390,22 @@ verify-generated: bundle generate
 
 install-git-hooks:
 	git config core.hooksPath .githooks	
+
+GOLANGCI_LINT := $(shell command -v golangci-lint 2> /dev/null)
+GOLANGCI_LINT_LOCAL_VERSION := $(shell command golangci-lint --version | awk '{print $$4}')
+golangci-lint:
+ifdef GOLANGCI_LINT
+	@echo "Found golangci-lint"
+ifeq ($(GOLANGCI_LINT_LOCAL_VERSION),$(GOLANGCI_LINT_VERSION))
+		@echo golangcilint using same version
+else
+		@echo "Mismatch version, installing golangci-lint $(GOLANGCI_LINT_VERSION)"
+		curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(BIN_DIR) $(GOLANGCI_LINT_VERSION)
```

Comment:
downloading binaries is ok-ish, but `curl | sh` is not unless 1. it's a critical piece AND 2. there are no other practical ways. Can we just curl the binary?
In addition, I'd download a tag/fixed release, not vs master

Reply:
Using `install.sh` is the recommended way to install golangci-lint https://golangci-lint.run/usage/install/
`install.sh` script just checks OS, platform and arch and download the proper "tar.gz" with the binary.

It is not downloading the `master` but the version specified in `GOLANGCI_LINT_VERSION`. 
It is using the `master` version of the `install.sh` script but that can be change if we end using it and it is an issue

Reply:
Right about only fetching `install.sh` from master (even though it's still a moving target, is a bit less evil).
In out case we know exactly the OS, platform and arch, so I'd still unconvinced a `curl | shell` is worth the challenges.